elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java

```diff
@@ -42,6 +45,9 @@ public SQLReferenceVisitor(MetaDataStore metaDataStore, String tableAlias, SQLDi
      */
     @Override
     protected String visitPhysicalReference(String reference) {
+        if (reference.indexOf('$') == 0) {
+            reference = reference.substring(1);
+        }
         return getFieldAlias(applyQuotes(tableAliases.peek()), applyQuotes(reference));
```

Comment:
Instead of
```java
private final Stack<String> tableAliases
```
can we make it unsynchronized `Deque`?

Reply:
sure.